pandas/core/frame.py

```diff
@@ -2563,8 +2562,16 @@ def _box_item_values(self, key, values):
 
     def _box_col_values(self, values, items):
         """ provide boxed values for a column """
-        return self._constructor_sliced._from_array(values, index=self.index,
-                                                    name=items, fastpath=True)
+        # This check here was previously performed in Series._from_array
+        # By doing it here there is no need for that function anymore
+        # GH#19883.
+        from pandas.core.dtypes.generic import ABCSparseArray
```

Comment:
make this a function like ``_get_series_result_type`` and put it near ``_get_frame_result_type`` in ``pandas/core/dtypes/concat.py``, it should just take self and return the type.

Reply:
I see the idea.
Wouldn't it make more sense to be in `pandas/core/dtypes/commons.py` ?  As that file is already imported to `frame.py`?

Reply:
no, common is for introspection, not conversion

Reply:
ok!